signal/events_shhext.go

```diff
@@ -158,6 +157,6 @@ func SendWhisperFilterAdded(filters []*Filter) {
 	send(EventWhisperFilterAdded, WhisperFilterAddedSignal{Filters: filters})
 }
 
-func SendNewMessages(response *statusproto.MessengerResponse) {
-	send(EventNewMessages, response)
+func SendNewMessages(obj json.Marshaler) {
+	send(EventNewMessages, obj)
 }
```

Comment:
Can any `json.Marshaler` be sent, or only a specific subset of `json.Marshaler`?

Reply:
any could be sent, as long as it explicitly implement `MarshalJSON`

Reply:
My question was more, *should* the `obj` be limited to a subset of `json.Marshaler` incase we wanted to control what exactly could be sent here. Something like:

```go
type SendAbleEventMessage interface {
  json.Marshaler
  SomeOtherMethodUniqueToThisTypeOfStruct()
}
```

But on thinking about it maybe `SendNewMessages()` should be very general and can received anything, and we control what is sent upstream. I may be over thinking this one.

Reply:
Ah, I think it's fine for now, we only send `MessengerResponse` and I don't think we will extend it much, the only reason it's made generic is that otherwise I was getting into issues with circular dependencies :)